src/plugins/intel_gpu/src/runtime/execution_config.cpp

```diff
@@ -195,7 +194,7 @@ void ExecutionConfig::apply_debug_options(const cldnn::device_info& info) {
     }
 
     GPU_DEBUG_IF(debug_config->enable_dynamic_quantize) {
-        set_property(ov::hint::dynamic_quantization_group_size(UINT64_MAX));
+        set_property(ov::hint::dynamic_quantization::group_size(ov::hint::dynamic_quantization::PER_TOKEN));
```

Comment:
looks confused that group size has value PER_TOKEN
size is numerical value IMO

Reply:
Are you suggesting to have something like?
```
hint::dynamic_quantization::mode = DISABLED / PER_TOKEN / GROUPED
hint::dynamic_quantization::group_size = group_size is valid only when mode is GROUPED.
```
I don't have objection here but I heard NNCF is using similar approach to represent PER_TOKEN as UINT64_MAX.

Could you share your opinion @vladimir-paramuzov ?

Reply:
Maybe something like this?
```cpp
namespace dynamic_quantization {
struct Mode  {
    constexpr operator uint64_t() const {
        return gs_num;
    }

    uint64_t gs_num = 0;

    static constexpr Mode DISABLED() { return Mode(0); }
    static constexpr Mode PER_TOKEN() { return Mode(UINT64_MAX); }
    static constexpr Mode GROUPED(size_t group) { return Mode(group); }

private:
    constexpr Mode() : gs_num{0} {};

    constexpr Mode(const uint64_t num_) : gs_num{num_} {}
};

static constexpr Property<Mode, ov::PropertyMutability::RW> mode{"DYNAMIC_QUANTIZATION_MODE"};

} // namespace dynamic_quantization

core.set_property(ov::hint::dynamic_quantization::mode(ov::hint::dynamic_quantization::Mode::DISABLED()));
core.set_property(ov::hint::dynamic_quantization::mode(ov::hint::dynamic_quantization::Mode::GROUPED(42)));
core.set_property(ov::hint::dynamic_quantization::mode(ov::hint::dynamic_quantization::Mode::PER_TOKEN()));
```

`gs_num`  can also be private and we may probably have explicit getter.

Reply:
@vladimir-paramuzov's proposal looks better

Reply:
I see. Thanks for the comment. I'll update accordingly.